governance/proposals/012-set-protocol-fee-cross-bridge.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * This proposal send calls accross the bridge to enable protocol fees
+ * on all supported networks.
+ */
+const { ADDRESS_ZERO, getNetwork } = require('@unlock-protocol/hardhat-helpers')
+const { Unlock } = require('@unlock-protocol/contracts')
+const { IConnext, targetChains } = require('../helpers/bridge')
+const { ethers } = require('hardhat')
+const PROTOCOL_FEE_IN_BASIS_POINTS = '100' // 1% in basis points
+
+const parseSetProtocolFeeCalls = async (destChainId) => {
+  const { unlockAddress } = await getNetwork(destChainId)
+
+  console.log(`Proposol to set protocolFee to ${ethers.formatEther(protocolFee)}
+    - unlock : ${unlockAddress}`)
+
+  // get Unlock interface
+  const { interface: unlockInterface } = await ethers.getContractAt(
+    Unlock.abi,
+    unlockAddress
+  )
+
+  const calldata = unlockInterface.encodeFunctionData('setProtocolFee', [
+    PROTOCOL_FEE_IN_BASIS_POINTS,
+  ])
+
+  // encode instructions to be executed by the SAFE
+  const moduleData = await ethers.defaultAbiCoder.encode(
+    ['address', 'uint256', 'bytes', 'bool'],
+    [
+      unlockAddress, // to
+      0, // value
+      calldata, // data
+      1, // operation: 0 for CALL, 1 for DELEGATECALL
+    ]
+  )
+
+  const oldProtocolFee = await unlockInterface.protocolFee()
+  const explainers = [
+    `- Protocol fee for ${unlockAddress} set to ${PROTOCOL_FEE_IN_BASIS_POINTS}
+  \`setProtocolFee(${PROTOCOL_FEE_IN_BASIS_POINTS})\` (Previous Protocol Fee: ${oldProtocolFee})`,
+  ]
+
+  return {
+    explainers,
+    moduleData,
+  }
+}
+
+const parseBridgeCall = async ({ destChainId, moduleData }) => {
+  const { governanceBridge } = await getNetwork(destChainId)
+
+  // get bridge info on receiving chain
+  const {
+    domainId: destDomainId,
+    modules: { connextMod: destAddress },
+  } = governanceBridge
+
+  // get bridge address on mainnet
+  const {
+    governanceBridge: { connext: bridgeAddress },
+  } = await getNetwork(1)
+
+  if (!destDomainId || !destAddress) {
+    throw Error('Missing bridge information')
+  }
+
+  // parse call for bridge
+  return {
+    contractAddress: bridgeAddress,
+    contractNameOrAbi: IConnext,
+    functionName: 'xcall',
+    functionArgs: [
+      destDomainId,
+      destAddress, // destMultisigAddress,
+      ADDRESS_ZERO, // asset
+      ADDRESS_ZERO, // delegate
+      0, // amount
+      30, // slippage
+      moduleData, // calldata
+    ],
+  }
+}
+
+module.exports = async () => {
+  const explainers = []
+
+  // get protocol fee call for mainnet
+  console.log(`Parsing for Ethereum Mainnet (1)`)
+  const { calls: mainnetCalls, explainers: mainnetExplainers } =
+    await parseSetProtocolFeeCalls(1)
```

Comment:
the mainnet call needs to be parsed 
```suggestion
  const { moduleData: mainnetModuleData, explainers: mainnetExplainers } =
    await parseSetProtocolFeeCalls(1)
  const mainnetCall = { calldata: mainnetModuleData, contractAddress: unlockAddress}
  
```

Reply:
the call are direct (no bridge) so need to specify the address of the targert contract

Reply:
Can you please shed more light on what you mean here?

Reply:
> Can you please shed more light on what you mean here?

The mainnet call to Unlock is not going through the bridge, so there is no need to parse it as a "moduleData" (which is required by a Safe multisig).